core/mm/core_mmu.c

```diff
@@ -1081,12 +1080,22 @@ static void collect_mem_ranges(struct memory_map *mem_map)
 {
 	const struct core_mmu_phys_mem *mem = NULL;
 	vaddr_t ram_start = secure_only[0].paddr;
+	size_t n = 0;
 
 #define ADD_PHYS_MEM(_type, _addr, _size) \
 		add_phys_mem(mem_map, #_addr, (_type), (_addr), (_size))
 
 	if (IS_ENABLED(CFG_CORE_RWDATA_NOEXEC)) {
-		ADD_PHYS_MEM(MEM_AREA_TEE_RAM_RO, ram_start,
+		paddr_t next_pa = 0;
+
+		/*
+		 * Read-only and read-execute physical memory areas must
+		 * not be mapped by MEM_AREA_SEC_RAM_OVERALL, but all the
+		 * read/write should.
+		 */
+		ADD_PHYS_MEM(MEM_AREA_TEE_RAM, ram_start,
+			     VCORE_UNPG_RX_PA - ram_start);
+		ADD_PHYS_MEM(MEM_AREA_SEC_RAM_OVERALL, ram_start,
```

Comment:
Would it make more sense only to map the gap between `TEE_RAM_START` and `TEE_LOAD_ADDR` in `MEM_AREA_SEC_RAM_OVERALL`?

Reply:
I think it would, unless some platforms load a TEE core image made on core binary image prepend with some data expected to be read-only mapped. Could be strange but current implementation allows that.
Maybe this old scheme could be deprecated. In that case, at least the commit message should explicitly state it IMHO.